### PR TITLE
Fix calculation of access rights when allow_introspection_functions = 0

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Bug Fix

Changelog entry:
Fix calculation of access rights when allow_introspection_functions=0

### CODE REVIEW

File: tests/integration/test_settings_profile/test.py [hunk 1]
```diff
@@ -166,21 +166,11 @@ def test_show_profiles():
 
 def test_allow_introspection():
     assert "Not enough privileges" in instance.query_and_get_error("SELECT demangle('a')", user="robin")
-    
-    instance.query("GRANT ALL ON *.* TO robin")
-    assert "Introspection functions are disabled" in instance.query_and_get_error("SELECT demangle('a')", user="robin")
-
-    instance.query("ALTER USER robin SETTINGS allow_introspection_functions=1")
```
Comment (abyss7) [inline tests/integration/test_settings_profile/test.py:173]: Why have you shrinked the test that much? Why don't you test previous scenarios?

Comment (vitlibar) [reply tests/integration/test_settings_profile/test.py:185]: That's good point. I've changed the test to keep some of the previous checks.

**Comment by bgranvea** [comment]
does this PR fix this kind of errors?

> GRANT ALL ON `xxx`.* TO `yyyy`
Clickhouse error when executing command: Code: 446, Message: default: Introspection functions are disabled

(with Clickhouse 20.6.3.28)

**Comment by vitlibar** [comment]
> does this PR fix this kind of errors?
> 
> > GRANT ALL ON `xxx`.* TO `yyyy`
> > Clickhouse error when executing command: Code: 446, Message: default: Introspection functions are disabled
> 
> (with Clickhouse 20.6.3.28)

No, it hasn't fixed yet.
To use the `ALL` while you're granting privileges you should set allow_introspection_functions=1 (because all privileges include INTROSPECTION). For now you can either set allow_introspection_functions=1 for the user who grants access rights or use a statement like `GRANT SELECT, INSERT, ...` instead of `GRANT ALL`.

**Comment by vitlibar** [comment]
> does this PR fix this kind of errors?
> 
> > GRANT ALL ON `xxx`.* TO `yyyy`
> > Clickhouse error when executing command: Code: 446, Message: default: Introspection functions are disabled
> 
> (with Clickhouse 20.6.3.28)

I'm fixing that error in https://github.com/ClickHouse/ClickHouse/pull/13987

**Pull request merged**